src/yaml-helpers.h

```diff
@@ -41,7 +41,7 @@
 }
 #define YAML_SCALAR_PLAIN(event_ptr, emitter_ptr, scalar) \
 { \
-    yaml_scalar_event_initialize(event_ptr, NULL, (yaml_char_t *)YAML_STR_TAG, (yaml_char_t *)scalar, strlen(scalar), 1, 0, YAML_PLAIN_SCALAR_STYLE); \
+    yaml_scalar_event_initialize(event_ptr, NULL, (yaml_char_t *)YAML_STR_TAG, (yaml_char_t *)scalar, (int)strlen(scalar), 1, 0, YAML_PLAIN_SCALAR_STYLE); \
```

Comment:
question: Do we need some assert/checks to make sure long strings (`size_t`) do not overflow our `int` datatype?

Reply:
nitpick (non-blocking): I guess we could also just pass `-1` as the `length` parameter and letting `libyaml` deal with it itself, avoiding any extra calls. But it's fine as-is, too.

See: https://github.com/yaml/libyaml/pull/102

Reply:
hm I see. That seems to be an undocumented behavior though? The doc string doesn't mention it https://github.com/yaml/libyaml/blob/0.2.5/include/yaml.h#L580

Reply:
Yeah, I though about that, too. So we might be safer keeping our code as-is.